### ISSUE TITLE
test_proc_cmdline_get_bool fails when Secure Boot is enabled

### ISSUE THREAD

**Issue by openmindead** [systemd/systemd#14864]
**systemd version the issue has been seen with**
> v244.2, v244.3 (no issue observed when building v242, v243.6 or 244.1, for instance)

**Used distribution**
 > Manjaro Linux

**Expected behaviour you didn't see**
```
291/485 test-proc-cmdline                       OK       0.57 s 

--- command ---
09:00:39 PATH='/home/openm/Git/systemd/src/build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/kde/src/kdesrc-build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/kde/src/kdesrc-build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/home/openm/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/var/lib/snapd/snap/bin' SYSTEMD_KBD_MODEL_MAP='/home/openm/Git/systemd/src/systemd-stable/src/locale/kbd-model-map' SYSTEMD_LANGUAGE_FALLBACK_MAP='/home/openm/Git/systemd/src/systemd-stable/src/locale/language-fallback-map' /home/openm/Git/systemd/src/build/test-proc-cmdline
--- stderr ---
/* test_proc_cmdline_parse */
kernel cmdline option <initrd> = <intel-ucode.img>
kernel cmdline option <initrd> = <initramfs-5.5-x86_64.img>
kernel cmdline option <ro> = <n/a>
kernel cmdline option <quiet> = <n/a>
kernel cmdline option <splash> = <n/a>
kernel cmdline option <audit> = <1>
kernel cmdline option <apparmor> = <1>
kernel cmdline option <security> = <apparmor>
kernel cmdline option <nouveau.noaccel> = <1>
kernel cmdline option <nowatchdog> = <n/a>
kernel cmdline option <random.trust_cpu> = <on>
kernel cmdline option <rootfstype> = <ext4>
kernel cmdline option <root> = <UUID=4c74074d-5543-4bea-89a2-0f96a039ccfe>
kernel cmdline option <resume> = <UUID=4c74074d-5543-4bea-89a2-0f96a039ccfe>
kernel cmdline option <resume_offset> = <6762496>
/* test_proc_cmdline_override */
/* test_proc_cmdline_given (flip: no) */
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
/* test_proc_cmdline_given (flip: yes) */
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <zumm> = <n/a>
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <rd.zumm> = <n/a>
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
/* test_proc_cmdline_key_streq */
/* test_proc_cmdline_key_startswith */
/* test_proc_cmdline_get_key */
/* test_proc_cmdline_get_bool */
/* test_proc_cmdline_get_key_many */
-------
```

**Unexpected behaviour you saw**
```
291/485 test-proc-cmdline                       FAIL     0.87 s (killed by signal 6 SIGABRT)

--- command ---
09:15:22 SYSTEMD_KBD_MODEL_MAP='/home/openm/Git/systemd/src/systemd-stable/src/locale/kbd-model-map' SYSTEMD_LANGUAGE_FALLBACK_MAP='/home/openm/Git/systemd/src/systemd-stable/src/locale/language-fallback-map' PATH='/home/openm/Git/systemd/src/build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/kde/src/kdesrc-build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/kde/src/kdesrc-build:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/opt/cuda/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/openm/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/home/openm/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/var/lib/snapd/snap/bin' /home/openm/Git/systemd/src/build/test-proc-cmdline
--- stderr ---
/* test_proc_cmdline_parse */
kernel cmdline option <ro> = <n/a>
kernel cmdline option <quiet> = <n/a>
kernel cmdline option <splash> = <n/a>
kernel cmdline option <audit> = <1>
kernel cmdline option <apparmor> = <1>
kernel cmdline option <security> = <apparmor>
kernel cmdline option <nouveau.noaccel> = <1>
kernel cmdline option <nowatchdog> = <n/a>
kernel cmdline option <random.trust_cpu> = <on>
kernel cmdline option <rootfstype> = <ext4>
kernel cmdline option <root> = <UUID=4c74074d-5543-4bea-89a2-0f96a039ccfe>
kernel cmdline option <resume> = <UUID=4c74074d-5543-4bea-89a2-0f96a039ccfe>
kernel cmdline option <resume_offset> = <6762496>
/* test_proc_cmdline_override */
/* test_proc_cmdline_given (flip: no) */
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
/* test_proc_cmdline_given (flip: yes) */
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <zumm> = <n/a>
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
parse_item_given: option <foo_bar> = <quux>
parse_item_given: option <wuff-piep> = <tuet >
parse_item_given: option <rd.zumm> = <n/a>
parse_item_given: option <space> = <x y z>
parse_item_given: option <miepf> = <uuu>
/* test_proc_cmdline_key_streq */
/* test_proc_cmdline_key_startswith */
/* test_proc_cmdline_get_key */
/* test_proc_cmdline_get_bool */
Assertion 'proc_cmdline_get_bool("foo_bar", &value) > 0 && value == true' failed at src/test/test-proc-cmdline.c:179, function test_proc_cmdline_get_bool(). Aborting.
-------
```

**Steps to reproduce the problem**
Enable Secure Boot (in my case I use my own keys in db to sign systemd-boot and unified kernel images, not sure if the same would happen when using shim + grub) and try to build systemd.
[meson-log_fail.txt](https://github.com/systemd/systemd/files/4190983/meson-log_fail.txt)
[meson-log_success.txt](https://github.com/systemd/systemd/files/4190984/meson-log_success.txt)
[testlog_fail.txt](https://github.com/systemd/systemd/files/4190985/testlog_fail.txt)
[testlog_success.txt](https://github.com/systemd/systemd/files/4190990/testlog_success.txt)

**Comment by openmindead** [comment]
Also I tried to build 245-rc1 using PKBUILD provided by AUR, same thing happened.

**Comment by openmindead** [comment]
Reproducible with shim + signed grub + signed vmlinuz + unsigned initrds.

**Comment by openmindead** [comment]
This commit is the culprit: https://github.com/systemd/systemd/commit/c7d26acce6dcb0e72be6160873fac758e9b7c440
Specifically -- changes in `src/basic/proc-cmdline.c`.

**Comment by arianvp** [comment]
Thanks for the ping. I will have a look this weekend and try to fix. is this commit already in a release; and if so should  I also open a backport PR?

**Comment by arianvp** [comment]
I think the test only needs to be adjusted. as it's not aware of the fact that `SYSTEMD_EFI_OPTIONS` is not read when SecureBoot is enabled

**Comment by openmindead** [comment]
@arianvp yep, it happens only when running after-build tests on a machine with SB enabled.

**Comment by poettering** [comment]
Fix waiting in #15640. ptal.

**Comment by openmindead** [comment]
@poettering I tried to build a package with your changes, no error occurred.
So this peculiar issue may be considered resolved.
